dask_sql/physical/utils/statistics.py

```diff
@@ -0,0 +1,207 @@
+from dask.utils_test import hlg_layer
+
+from dask_sql.utils import make_pickable_without_dask_sql
+
+logger = logging.getLogger(__name__)
+
+
+def parquet_statistics(
+    ddf: dd.DataFrame,
+    columns: List | None = None,
+    parallel: int | False | None = None,
+    **compute_kwargs,
+) -> List[dict] | None:
+    """Extract Parquet statistics from a Dask DataFrame collection
+
+    WARNING: This API is experimental
+
+    Parameters
+    ----------
+    ddf
+        Dask-DataFrame object to extract Parquet statistics from.
+    columns
+        List of columns to collect min/max statistics for. If ``None``
+        (the default), only 'num-rows' statistics will be collected.
+    parallel
+        The number of distinct files to collect statistics for
+        within a distinct ``dask.delayed`` task. If ``False``, all
+        statistics will be parsed on the client process. If ``None``,
+        the value will be set to 16 for remote filesystem (e.g s3)
+        and ``False`` otherwise. Default is ``None``.
+    **compute_kwargs
+        Key-word arguments to pass through to ``dask.compute`` when
+        ``parallel`` is not ``False``.
+
+    Returns
+    -------
+    statistics
+        List of Parquet statistics. Each list element corresponds
+        to a distinct partition in ``ddf``. Each element of
+        ``statistics`` will correspond to a dictionary with
+        'num-rows' and 'columns' keys::
+
+            ``{'num-rows': 1024, 'columns': [...]}``
+
+        If column statistics are available, each element of the
+        list stored under the "columns" key will correspond to
+        a dictionary with "name", "min", and "max" keys::
+
+            ``{'name': 'col0', 'min': 0, 'max': 100}``
+    """
+
+    # Check that we have a supported `ddf` object
+    if not isinstance(ddf, dd.DataFrame):
+        raise ValueError(f"Expected Dask DataFrame, got {type(ddf)}.")
+
+    # Be strict about columns argument
+    if columns:
+        if not isinstance(columns, list):
+            raise ValueError(f"Expected list, got columns={type(columns)}.")
+        elif not set(columns).issubset(set(ddf.columns)):
+            raise ValueError(f"columns={columns} must be a subset of {ddf.columns}")
+
+    # Extract "read-parquet" layer from ddf
+    try:
+        layer = hlg_layer(ddf.dask, "read-parquet")
+    except KeyError:
+        layer = None
+
+    # Make sure we are dealing with a
+    # ParquetFunctionWrapper-based DataFrameIOLayer
+    if not isinstance(layer, DataFrameIOLayer) or not isinstance(
+        layer.io_func, ParquetFunctionWrapper
+    ):
+        logger.warning(
+            f"Could not extract Parquet statistics from {ddf}."
+            f"\nAttempted IO layer: {layer}"
+        )
+        return None
+
+    # Collect statistics using layer information
+    parts = layer.inputs
+    fs = layer.io_func.fs
+    engine = layer.io_func.engine
+    if not isinstance(engine(), ArrowDatasetEngine):
```

Comment:
Any reason we prefer instantiating `engine` over doing something like

```suggestion
    if not issubclass(engine, ArrowDatasetEngine):
```

Reply:
Oops - Forgot about `issubclass`. That's much better here (to capture a cudf-based engine as well)